src/source/Ice/NatBehaviorDiscovery.c

```diff
@@ -0,0 +1,404 @@
+/**
+ * Kinesis Video Producer Ice Utils
+ */
+#define LOG_CLASS "NatTypeDiscovery"
+#include "../Include_i.h"
+
+#define DEFAULT_NAT_TEST_MAX_BINDING_REQUEST_COUNT                  5
+#define DEFAULT_TEST_NAT_TEST_RESPONSE_WAIT_TIME                    500 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND
+
+#define NAT_BEHAVIOR_NONE_STR                                       (PCHAR) "NONE"
+#define NAT_BEHAVIOR_NOT_BEHIND_ANY_NAT_STR                         (PCHAR) "NOT_BEHIND_ANY_NAT"
+#define NAT_BEHAVIOR_NO_UDP_CONNECTIVITY_STR                        (PCHAR) "NO_UDP_CONNECTIVITY"
+#define NAT_BEHAVIOR_ENDPOINT_INDEPENDENT_STR                       (PCHAR) "ENDPOINT_INDEPENDENT"
+#define NAT_BEHAVIOR_ADDRESS_DEPENDENT_STR                          (PCHAR) "ADDRESS_DEPENDENT"
+#define NAT_BEHAVIOR_PORT_DEPENDENT_STR                             (PCHAR) "PORT_DEPENDENT"
+
+typedef struct {
+    /* Should be able to contain max number of binding response we can get */
+    PStunPacket bindingResponseList[DEFAULT_NAT_TEST_MAX_BINDING_REQUEST_COUNT * 6];
```

Comment:
Why 6? Add a constant like MAX_ENUM_VALUE in the enum

Reply:
fixed